src/components/AddBook.js

```diff
@@ -0,0 +1,39 @@
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+
+import { addBook } from '../redux/books/books';
+
+const AddBook = () => {
+  const [book, setBook] = useState({});
+
+  const dispatch = useDispatch();
+
+  const inputValue = (element) => {
+    setBook({
+      ...book,
+      id: uuidv4(),
+      [element.target.name]: element.target.value,
+    });
+  };
+
+  return (
+    <div className="form-container">
+      <h3 className="add-book-title">ADD NEW BOOK</h3>
+      <form
+        action="/action_page.php"
+        onSubmit={(e) => {
+          e.preventDefault();
+          dispatch(addBook(book));
+          e.target.reset();
+        }}
+      >
```

Comment:
- [ ] Please know that it's a better practice to create the function that handles the dispatch outside of the return clause and then call it from here, kindly adjust that.

Reply:
Fixed.